symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java

```diff
@@ -0,0 +1,330 @@
+  private JsonMapper jsonMapper;
+
+  @Before
+  public void initBot() {
+    final SymConfig symConfig = SymConfigLoader.loadConfig("src/test/resources/sym-config.json");
+
+    this.symBotClient = Mockito.mock(SymBotClient.class);
+    Mockito.when(this.symBotClient.getConfig()).thenReturn(symConfig);
+
+    this.templateService = Mockito.mock(TemplateService.class);
+    this.jsonMapper = Mockito.mock(JsonMapper.class);
+
+    this.messagesClient = Mockito.mock(MessagesClient.class);
+    Mockito.when(this.symBotClient.getMessagesClient()).thenReturn(this.messagesClient);
+
+    this.messageClientImpl = new MessageClientImpl(this.symBotClient, this.templateService, this.jsonMapper);
+
+    this.messageClientImplMocked = Mockito.mock(MessageClientImpl.class);
+  }
+
+  @Test
+  public void testSendMessageSymphonyMessage() throws SymphonyClientException {
+    final String streamId = "streamId";
+    this.testSendMessage(streamId, true);
+    this.testSendMessage(streamId, false);
+  }
+
+  @Test
+  public void testSendMessageJsonData() throws SymphonyClientException {
+    final String streamId = "streamId";
+    final String message = "This is a message";
+    final String jsonData = "{\"message\": \"This is a message\"}";
+    this.messageClientImpl.sendMessage(streamId, message, jsonData);
+  }
+
+  @Test
+  public void testSendMessageJsonDataWithAttachments() throws SymphonyClientException {
+    final String streamId = "streamId";
+    final String message = "This is a message";
+    final String jsonData = "{\"message\": \"This is a message\"}";
+    final List<MessageAttachmentFile> messageAttachmentFiles = this.initMessageAttachmentFiles();
+    this.messageClientImpl.sendMessage(streamId, message, jsonData, messageAttachmentFiles);
+  }
+
+  @Test(expected = SymphonyClientException.class)
+  public void testSendMessageJsonDataWithAttachmentsWithException() throws SymphonyClientException {
+    final String streamId = "streamId";
+    final String message = "This is a message";
+    final String jsonData = "{\"message\": \"This is a message\"}";
+    final List<MessageAttachmentFile> messageAttachmentFiles = this.initMessageAttachmentFiles();
+
+    Mockito.when(this.symBotClient.getMessagesClient().sendMessage(anyString(), any(OutboundMessage.class)))
+        .thenThrow(SymClientException.class);
+
+    this.messageClientImpl.sendMessage(streamId, message, jsonData, messageAttachmentFiles);
+  }
+
+  @Test
+  public void testDownloadMessageAttachments() throws SymphonyClientException {
+    final MessageEvent messageEvent = this.initMessageEvent();
+    final byte[] fileContent1 = new byte[] {(byte) 6789, (byte) 47, (byte) 567};
+    final byte[] fileContent2 = new byte[] {(byte) 990, (byte) 467, (byte) 5654};
+    final byte[] fileContent3 = new byte[] {(byte) 687, (byte) 479, (byte) 574};
+    final List<FileAttachment> fileAttachmentList = this.initFileAttachmentsList(fileContent1, fileContent2, fileContent3);
```

Comment:
Why not reuse `this.initMessageAttachmentFiles()` ?

Reply:
I don't have to init a `List<MessageAttachmentFile>`, but I have to init a `List<FileAttachment>` here. I don't see how you want me to do so

Reply:
Maybe I create a method `initFileAttachmentList` ?

Reply:
Maybe I put this code : 
```
    final byte[] fileContent1 = new byte[] {(byte) 6789, (byte) 47, (byte) 567};
    final byte[] fileContent2 = new byte[] {(byte) 990, (byte) 467, (byte) 5654};
    final byte[] fileContent3 = new byte[] {(byte) 687, (byte) 479, (byte) 574};
```
in initFileAttachmentsList

Reply:
No, i can't do this

Reply:
Sorry, types are different. This is so misleading.